### PR TITLE
Pull Requests: Add update / remove delete

### LINKED ISSUE #8202: Pull Requests MVP: Update/Delete Pull Request
Reference: https://github.com/treeverse/lakeFS/issues/8166

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #8202 

## Change Description

Add UpdatePullRequest, remove support for delete

### Testing Details

Added unit tests

### CODE REVIEW

File: pkg/catalog/fake_graveler_test.go [hunk 1]
```diff
@@ -314,6 +314,14 @@ func (g *FakeGraveler) ListPullRequests(context.Context, *graveler.RepositoryRec
 	panic("implement me")
 }
 
+func (g *FakeGraveler) DeletePullRequest(context.Context, *graveler.RepositoryRecord, graveler.PullRequestID) error {
```
Comment (itaigilo) [inline pkg/catalog/fake_graveler_test.go:317]: I think it's not needed anymore...

File: pkg/graveler/graveler.go [hunk 2]
```diff
@@ -748,6 +755,9 @@ type Collaborator interface {
 
 	// ListPullRequests lists pull requests on repository
 	ListPullRequests(ctx context.Context, repository *RepositoryRecord) (PullsIterator, error)
+
+	// UpdatePullRequest delete pull request from the given repository
```
Comment (itaigilo) [inline pkg/graveler/graveler.go:759]: >// UpdatePullRequest update pull request in the given repository

File: pkg/graveler/validate.go [hunk 3]
```diff
@@ -88,6 +88,9 @@ func ValidatePullRequestID(v interface{}) error {
 
 	// TODO (niro): Any other validations?
```
Comment (itaigilo) [inline pkg/graveler/validate.go:89]: This TODO is done and can be removed?

**Review by itaigilo — APPROVED** [review]
LGTM.

Only some nit.

File: pkg/graveler/validate.go (hunk 3 above)
Comment (N-o-Z) [reply pkg/graveler/validate.go:89]: No - still relevant

File: pkg/graveler/graveler.go (hunk 2 above)
Comment (N-o-Z) [reply pkg/graveler/graveler.go:759]: Please use github suggestions next time - it'll make it easier to update

Comment (N-o-Z) [reply pkg/graveler/graveler.go]: Please use github suggestions next time - it'll make it easier to update

**Pull request merged**